pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -44,6 +45,11 @@ func (a *index) ServicesCollection(
 			}
 		}
 		waypoint := fetchWaypoint(ctx, Waypoints, Namespaces, s.ObjectMeta)
+		if waypoint != nil && !waypointForService(waypoint) {
+			// Waypoint does not support Service traffic, no more work to do
+			log.Debugf("Skipping waypoint %s/%s; traffic type %s not supported", waypoint.Namespace, waypoint.Name, waypoint.TrafficType)
+			return nil
```

Comment:
This stops us from recognizing the existence of the service at all. I don't think that is intended. We should still recognize the service _just_ not set the invalid waypoint for the service.

Same comment applies for the other types so I won't repeat it a bunch of times.